saleor/product/models/base.py

```diff
@@ -201,6 +202,12 @@ def get_attribute(self, pk):
     def set_attribute(self, pk, value_pk):
         self.attributes[smart_text(pk)] = smart_text(value_pk)
 
+    def get_price_range(self, **kwargs):
+        if not self.variants.exists():
+            return PriceRange(self.price, self.price)
```

Comment:
This needs to mimic the discount logic from `ProductVariant.get_unit_price` (even if it may seem silly to care about discounts for unavailable products)